internal/postgres/config.go

```diff
@@ -322,52 +355,53 @@ done
-		// However, unlike the PGDATA directory, Postgres will _not_ error out
-		// if the permissions are wrong on the tablespace directory.
-		// Instead, when a tablespace is created in Postgres, Postgres will `chmod` the
-		// tablespace directory to match permissions on the PGDATA directory (either 700 or 750).
-		// Postgres setting the tablespace directory permissions:
-		// - https://git.postgresql.org/gitweb/?p=postgresql.git;f=src/backend/commands/tablespace.c;hb=REL_14_0#l600
-		// Postgres choosing directory permissions:
-		// - https://git.postgresql.org/gitweb/?p=postgresql.git;f=src/common/file_perm.c;hb=REL_14_0#l27
-		// Note: This permission change seems to happen only when the tablespace is created in Postgres.
-		// If the user manually `chmod`'ed the directory after the creation of the tablespace, Postgres
-		// would not attempt to change the directory permissions.
-		// Note: as noted below, we mount the tablespace directory to the mountpoint `/tablespaces/NAME`,
-		// and so we add the subdirectory `data` in order to set the permissions.
-		checkInstallRecreateCmd := strings.Join([]string{
-			`if [[ ! -e "${tablespace_dir}" || -O "${tablespace_dir}" ]]; then`,
-			`install --directory --mode=0750 "${tablespace_dir}"`,
-			`elif [[ -w "${tablespace_dir}" && -g "${tablespace_dir}" ]]; then`,
-			`recreate "${tablespace_dir}" '0750'`,
-			`else (halt Permissions!); fi ||`,
-			`halt "$(permissions "${tablespace_dir}" ||:)"`,
-		}, "\n")
+	mkdirs := make([]string, 0, 9+len(instance.TablespaceVolumes))
+	mkdirs = append(mkdirs, `dataDirectory "${postgres_data_directory}" || halt "$(permissions "${postgres_data_directory}" ||:)"`)
 
+	// If the user requests tablespaces, we want to make sure the directories exist with the correct owner and permissions.
+	//
+	// The path for tablespaces volumes is /tablespaces/NAME/data -- the `data` directory is so we can arrange the permissions.
+	if feature.Enabled(ctx, feature.TablespaceVolumes) {
 		for _, tablespace := range instance.TablespaceVolumes {
-			// The path for tablespaces volumes is /tablespaces/NAME/data
-			// -- the `data` path is added so that we can arrange the permissions.
-			tablespaceCmd = tablespaceCmd + "\ntablespace_dir=/tablespaces/" + tablespace.Name + "/data" + "\n" +
-				checkInstallRecreateCmd
+			dir := shell.QuoteWord("/tablespaces/" + tablespace.Name + "/data")
+			mkdirs = append(mkdirs, `dataDirectory `+dir+` || halt "$(permissions `+dir+` ||:)"`)
 		}
 	}
 
+	// Postgres creates "log_directory" but does *not* create any of its parent directories.
+	// Postgres omits the group-write S_IWGRP permission on the directory. Do both here while being
+	// careful to *not* touch "data_directory" contents until after `initdb` or Patroni bootstrap.
+	if path.IsAbs(logDir) && !strings.HasPrefix(logDir, dataDir) {
+		mkdirs = append(mkdirs,
+			`(`+shell.MakeDirectories(dataMountPath, logDir)+`) ||`,
```

Comment:
```suggestion
	if path.IsAbs(logDir) && !strings.HasPrefix(logDir, dataDir) {
		mkdirs = append(mkdirs,
			`(`+shell.MakeDirectories(dataDir, logDir)+`) ||`,
```
? `dataDir` is `dataMountPath` right now, yeah?

Reply:
No, `dataDir == "/pgdata/pg16"` and `dataMountPath == "/pgdata"`.